### PR TITLE
Force repo config

### PR DESCRIPTION
**Pull request by poikilotherm**
@landreev said we better force people to configure everything, so hard to track bugs like https://github.com/IQSS/dataverse/issues/9309 don't happen as easily.

### CODE REVIEW

**Comment by landreev** [comment]
Thank you! Will add review shortly. Since the PR is ready, we don't really need a separate issue requesting this change, do we? (And same for the other PR for the "earlier than the earliest datestamp"?)

**Comment by poikilotherm** [comment]
Ok 🤫 Don't tell anyone we did not create an issue first.

File: xoai-data-provider/src/main/java/io/gdcc/xoai/dataprovider/repository/RepositoryConfiguration.java [hunk 1]
```diff
@@ -146,119 +211,240 @@ public ResumptionTokenFormat getResumptionTokenFormat() {
...
+        Granularity granularity = Granularity.Second;
+        ResumptionTokenFormat resumptionTokenFormat =
+                new SimpleResumptionTokenFormat().withGranularity(Granularity.Second);
+        String repositoryName;
+        String baseUrl;
+        Instant earliestDate;
+        DeletedRecord deleteMethod;
+        Integer maxListIdentifiers = 100;
+        Integer maxListSets = 100;
+        Integer maxListRecords = 100;
```
Comment (landreev) [inline xoai-data-provider/src/main/java/io/gdcc/xoai/dataprovider/repository/RepositoryConfiguration.java:243]: Probably a stupid question, but why do we need these default values hard-coded here? - considering that we have microprofile config defaults for these?

Comment (poikilotherm) [reply xoai-data-provider/src/main/java/io/gdcc/xoai/dataprovider/repository/RepositoryConfiguration.java:243]: This has been a hardcoded default before and is also recommended within the OAI PMH implementation guidelines, too.

Yes, Dataverse has its own defaults. But anyone looking to reuse this library might have a different opinion, etc.

I do not want to add another dependency on MPConfig to this library, so I think leaving the 100 in place is a reasonable compromise between defaults, backward compatibility and reusability.

Comment (landreev) [reply xoai-data-provider/src/main/java/io/gdcc/xoai/dataprovider/repository/RepositoryConfiguration.java:243]: In my defense, I did acknowledge early on it was a stupid question. (I was literally confusing what we do in the Dataverse code base and what you do in this library - sorry).

**Review by landreev — APPROVED** [review]
All seems prudent. Thank you.

**Comment by poikilotherm** [comment]
Crossing fingers, merging now!

**Pull request merged**